libs/core/src/main/java/org/elasticsearch/core/RestApiVersion.java

```diff
@@ -23,37 +23,40 @@ public enum RestApiVersion {
     public final byte major;
 
     private static final RestApiVersion CURRENT = V_8;
+    private static final RestApiVersion PREVIOUS = V_7;
 
     RestApiVersion(int major) {
         this.major = (byte) major;
     }
 
-    public RestApiVersion previous() {
-        return fromMajorVersion(major - 1);
+    public boolean matches(Predicate<RestApiVersion> restApiVersionFunctions) {
+        return restApiVersionFunctions.test(this);
     }
 
-    public boolean matches(Function<RestApiVersion, Boolean> restApiVersionFunctions) {
-        return restApiVersionFunctions.apply(this);
+    public static RestApiVersion current() {
+        return CURRENT;
     }
 
-    private static RestApiVersion fromMajorVersion(int majorVersion) {
-        return valueOf("V_" + majorVersion);
+    public static RestApiVersion previous() {
+        return PREVIOUS;
     }
 
     public static RestApiVersion minimumSupported() {
-        return current().previous();
-    }
-
-    public static RestApiVersion current() {
-        return CURRENT;
+        return V_7;
     }
 
-    public static Function<RestApiVersion, Boolean> equalTo(RestApiVersion restApiVersion) {
-        return r -> r.major == restApiVersion.major;
+    public static Predicate<RestApiVersion> equalTo(RestApiVersion restApiVersion) {
+        return switch (restApiVersion) {
+            case V_8 -> r -> r.major == V_8.major;
```

Comment:
If performance is a concern, these predicates could be created statically and then just referenced here?

Reply:
That won't make a difference performance wise. We're not capturing anything in these lambdas so they just compile as static methods anyway I think?

Reply:
I'm not sure that is true. I thought a new lambda would be created here on every invocation, regardless of what is captured or not.

Reply:
I think the JLS doesn't explicitly enforce it, but practically we shouldn't have any allocation here. At least on OpenJDK17 we eventually always get the same instances returned because the `invokedynamic` is bootstrapped into a static method handle. Using a constant compiles the same I think.